src/_pytest/capture.py

```diff
@@ -681,7 +681,8 @@ def resume(self):
         setattr(sys, self.name, self.tmpfile)
         self._state = "resumed"
 
-    def writeorg(self, data):
+    def writeorg(self, data: str) -> None:
+        data = data.decode(self._old.encoding)
         self._old.write(data)
```

Comment:
Maybe write to `self._old.buffer` instead of doing the `decode`? Although it might not exist in case of "nested" patching of `sys.stdout` and friends (e.g. if someone patches `sys.stdout` to a `io.StringIO`, which doesn't have an underlying buffer, and then tries to capture it -- probably not a proper thing to do).